src/platform_impl/linux/wayland/state.rs

```diff
@@ -148,6 +152,8 @@ impl WinitState {
             xdg_shell: XdgShell::bind(globals, queue_handle)?,
             xdg_activation: XdgActivationState::bind(globals, queue_handle).ok(),
 
+            layer_shell: LayerShell::bind(globals, queue_handle)?,
```

Comment:
Does this make winit fail on compositors without layer-shell support? Like Gnome.

Reply:
The way it's written -> yes.

Reply:
So definitely something that would need to be fixed here.

An application using this will want some way to recognize when X is being used instead of Wayland, and when Wayland is in use but the layer-shell protocol isn't available.

Reply:
Yeah, but I mean, the entire approach present here I don't like, because it complicates internal code without a reason for that. I'd also like to have a way to bind stuff lazily.